tensorflow/python/eager/function.py

```diff
@@ -2525,6 +2535,43 @@ def _convert_variables_to_tensors(self, args, kwargs):
     kwargs = {kw: ops.convert_to_tensor(x) for kw, x in kwargs.items()}
     return tuple(args), kwargs
 
+  def _convert_annotated_args_to_tensors(self, args, kwargs):
+    if self.input_signature is not None:
+      return
+
+    args = list(args)
+    for i, arg in enumerate(args):
```

Comment:
Optional, if you get the chance: it would be ideal to disentangle these blocks, and separate them into stages:
 1. one set of loops to find type annotations, if any, resulting in an arg_annos, and kwarg_annos
 2. another set of loops to convert arguments as necessary

That way the logic is easier to follow an verify, and you avoid the duplication of the "if annotation == ops.Tensor: convert" logic, which is expected to become more complex in time.